desc/objectives/utils.py

```diff
@@ -149,15 +149,39 @@ def recover(x_reduced):
         y1 = con.compute_unscaled(*xpi)
         y2 = con.target
         y1, y2 = np.broadcast_arrays(y1, y2)
+
+        # If the error is very large, likely want to error out as
+        # it probably is due to a real mistake instead of just numerical
+        # roundoff errors.
         np.testing.assert_allclose(
             y1,
             y2,
-            atol=3e-14,
-            rtol=3e-14,
+            atol=1e-6,
+            rtol=1e-1,
             err_msg="Incompatible constraints detected, cannot satisfy constraint "
             + f"{con}.",
         )
 
+        # else check with tighter tols and throw an error, these tolerances
+        # could be tripped due to just numerical roundoff or poor scaling between
+        # constraints, so don't want to error out but we do want to warn the user.
+        atol = 3e-14
+        rtol = 3e-14
+
+        try:
+            np.testing.assert_allclose(y1, y2, atol=atol, rtol=rtol)
+        except AssertionError:
```

Comment:
might be cleaner to use the existing error message:
```python
except AssertionError as e:
    warnings.warn(str(e) + "\n This may indicate incompatible constraints, or be due to floating point error.")